man/adoc/bpftrace.adoc

```diff
@@ -977,6 +977,131 @@ The following snippets create a map with key signature `(int64, string[16])` and
 @[(pid, comm)]++
 ----
 
+==== Hygienic macros
+
+Hygienic macros (usually just "macros") provide a way for you to structure your script. They can be
+useful when you want to factor out code into smaller, more understandable parts. Or if you want to
+share code between probes.
+
+At a high level, macros can be thought of as semantic aware text replacement. They accept (optional)
+variable and map arguments - the body of the macro may only access maps and variables passed in through
+the arguments (the hygiene) or literal expressions. The body of the macro is exactly one block expression.
+
+For example, these are valid usages of macros:
+
+----
+config = {
+    unstable_macro=1;
+}
+
+macro one() {
+  1
+}
+
+macro add_one($other_name) {
+  $other_name + 1
+}
+
+macro add_one_to_each($a, @b) {
+  $a + 1 + @b + 1
+}
+
+macro side_effect($x) {
+  print($x)
+}
+
+macro add_two($x) {
+  add_one($x) + 1
+}
+
+BEGIN {
+  print(one());
+
+  $a = 42;
+  print(add_one($a));
+
+  @b = 3;
+  print(add_one_to_each($a, @b));
+
+  side_effect(5)
+
+  print(add_two(1));
+}
+
+END {
+  clear(@b);
+}
+
+/*
+ * Sample output:
+ * Attaching 2 probes...
+ * 1
+ * 43
+ * 47
+ * 5
+ * 3
+ */
+----
+
+Some examples of invalid macro usage:
+
+----
+config = {
+    unstable_macro=1;
+}
+
+macro not_expression() {
+  $var = 1;                    // BAD: Not an expression
+}
+
+macro unhygienic_access() {
+  @x++;                        // BAD: @x not passed in
```

Comment:
Maybe remove the `;` because it is also not an expression ;) For this one and below